`error: 'exchange' is not a member of 'std'` when trying to build with development environments

When trying to build from source using the development environments, I run into the following error:

```
/raid/charlesb/git/rapidsai/ucxx
-- The C compiler identification is GNU 12.3.0
-- The CXX compiler identification is GNU 12.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /raid/charlesb/miniforge3/envs/ucxx-cuda120/bin/x86_64-conda-linux-gnu-cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /raid/charlesb/miniforge3/envs/ucxx-cuda120/bin/x86_64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- CPM: Using local package rmm@24.02.0
-- CPM: Using local package GTest@1.13.0
-- Found Python3: /raid/charlesb/miniforge3/envs/ucxx-cuda120/include/python3.10 (found version "3.10.13") found components: Development Development.Module Development.Embed 
-- Configuring done (5.7s)
-- Generating done (0.0s)
-- Build files have been written to: /raid/charlesb/git/rapidsai/ucxx/cpp/build
[8/34] Building CXX object python/CMakeFiles/ucxx_python.dir/src/python_future.cpp.o
FAILED: python/CMakeFiles/ucxx_python.dir/src/python_future.cpp.o 
/raid/charlesb/miniforge3/envs/ucxx-cuda120/bin/x86_64-conda-linux-gnu-c++ -DFMT_HEADER_ONLY=1 -DFMT_SHARED -DLIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_INFO -DSPDLOG_FMT_EXTERNAL -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CUDA -DTHRUST_HOST_SYSTEM=THRUST_HOST_SYSTEM_CPP -DUCXX_ENABLE_PYTHON -DUCXX_ENABLE_RMM -Ducxx_python_EXPORTS -I/raid/charlesb/git/rapidsai/ucxx/cpp/include -I/raid/charlesb/git/rapidsai/ucxx/cpp/python/include -I/raid/charlesb/git/rapidsai/ucxx/cpp/build/include -I/raid/charlesb/git/rapidsai/ucxx/cpp/build/python/include -I/raid/charlesb/git/rapidsai/ucxx/cpp/src -I/raid/charlesb/miniforge3/envs/ucxx-cuda120/include/rapids/libcudacxx -I/raid/charlesb/miniforge3/envs/ucxx-cuda120/include/rapids -isystem /raid/charlesb/miniforge3/envs/ucxx-cuda120/targets/x86_64-linux/include -isystem /raid/charlesb/miniforge3/envs/ucxx-cuda120/include/python3.10 -fvisibility-inlines-hidden -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /raid/charlesb/miniforge3/envs/ucxx-cuda120/include -O3 -DNDEBUG -std=gnu++17 -fPIC -pthread -MD -MT python/CMakeFiles/ucxx_python.dir/src/python_future.cpp.o -MF python/CMakeFiles/ucxx_python.dir/src/python_future.cpp.o.d -o python/CMakeFiles/ucxx_python.dir/src/python_future.cpp.o -c /raid/charlesb/git/rapidsai/ucxx/cpp/python/src/python_future.cpp
/raid/charlesb/git/rapidsai/ucxx/cpp/python/src/python_future.cpp: In member function 'virtual void* ucxx::python::Future::release()':
/raid/charlesb/git/rapidsai/ucxx/cpp/python/src/python_future.cpp:98:15: error: 'exchange' is not a member of 'std'
   98 |   return std::exchange(_handle, nullptr);
      |               ^~~~~~~~
[32/34] Building CXX object CMakeFiles/ucxx.dir/src/worker.cpp.o
ninja: build stopped: subcommand failed.
```

It seems like the `conda` solver being used to create the environment has some impact, so for sake of reproducibility the specific `mamba` version I used was:

```
mamba 1.5.4
conda 23.11.0
```

And the specific environment I pulled:

```
$ mamba env create -n ucxx-cuda120 -f conda/environments/all_cuda-120_arch-x86_64.yaml
...
$ conda env export -n ucxx-cuda120
name: ucxx-cuda120
channels:
  - dask/label/dev
  - rapidsai-nightly
  - conda-forge
dependencies:
  - _libgcc_mutex=0.1=conda_forge
  - _openmp_mutex=4.5=2_gnu
  - autoconf=2.71=pl5321h2b4cb7a_1
  - automake=1.16.5=pl5321ha770c72_0
  - aws-c-auth=0.7.8=h538f98c_2
  - aws-c-cal=0.6.9=h5d48c4d_2
  - aws-c-common=0.9.10=hd590300_0
  - aws-c-compression=0.2.17=h7f92143_7
  - aws-c-event-stream=0.3.2=h0bcb0bb_8
  - aws-c-http=0.7.14=hd268abd_3
  - aws-c-io=0.13.36=he0cd244_2
  - aws-c-mqtt=0.10.0=h35285c7_0
  - aws-c-s3=0.4.5=h0448019_0
  - aws-c-sdkutils=0.1.13=h7f92143_0
  - aws-checksums=0.1.17=h7f92143_6
  - aws-crt-cpp=0.25.0=h1bbe558_2
  - aws-sdk-cpp=1.11.210=h0853bfa_5
  - binutils=2.40=hdd6e379_0
  - binutils_impl_linux-64=2.40=hf600244_0
  - binutils_linux-64=2.40=hbdbef99_2
  - bokeh=3.3.2=pyhd8ed1ab_0
  - brotli-python=1.1.0=py310hc6cd4ac_1
  - bzip2=1.0.8=hd590300_5
  - c-ares=1.23.0=hd590300_0
  - c-compiler=1.7.0=hd590300_0
  - ca-certificates=2023.11.17=hbcca054_0
  - cachetools=5.3.2=pyhd8ed1ab_0
  - cffi=1.16.0=py310h2fee648_0
  - cfgv=3.3.1=pyhd8ed1ab_0
  - click=8.1.7=unix_pyh707e725_0
  - cloudpickle=3.0.0=pyhd8ed1ab_0
  - cmake=3.28.1=hcfe8598_0
  - colorama=0.4.6=pyhd8ed1ab_0
  - contourpy=1.2.0=py310hd41b1e2_0
  - cuda-cccl_linux-64=12.0.90=ha770c72_1
  - cuda-cudart=12.0.107=hd3aeb46_8
  - cuda-cudart-dev=12.0.107=hd3aeb46_8
  - cuda-cudart-dev_linux-64=12.0.107=h59595ed_8
  - cuda-cudart-static=12.0.107=hd3aeb46_8
  - cuda-cudart-static_linux-64=12.0.107=h59595ed_8
  - cuda-cudart_linux-64=12.0.107=h59595ed_8
  - cuda-nvcc-dev_linux-64=12.0.76=ha770c72_1
  - cuda-nvcc-impl=12.0.76=h59595ed_1
  - cuda-nvcc-tools=12.0.76=h59595ed_1
  - cuda-nvrtc=12.0.76=hd3aeb46_2
  - cuda-nvtx=12.0.76=h59595ed_1
  - cuda-python=12.0.0=py310hfb71131_3
  - cuda-version=12.0=hffde075_2
  - cudf=24.02.00a129=cuda12_py310_231215_g2cb8f3da3a_129
  - cupy=12.3.0=py310hfc31588_0
  - cxx-compiler=1.7.0=h00ab1b0_0
  - cython=3.0.6=py310hc6cd4ac_0
  - cytoolz=0.12.2=py310h2372a71_1
  - dask=2023.12.1a231215=py_g3882bc62_14
  - dask-core=2023.12.1a231214=py_g2d8f6d54c_27
  - dask-cuda=24.02.00a23=py310_231215_g0f34116_23
  - dask-cudf=24.02.00a129=cuda12_py310_231215_g2cb8f3da3a_129
  - distlib=0.3.8=pyhd8ed1ab_0
  - distributed=2023.12.1a231215=py_g3882bc62_14
  - distro=1.8.0=pyhd8ed1ab_0
  - dlpack=0.5=h9c3ff4c_0
  - exceptiongroup=1.2.0=pyhd8ed1ab_0
  - fastrlock=0.8.2=py310hc6cd4ac_1
  - filelock=3.13.1=pyhd8ed1ab_0
  - fmt=10.1.1=h00ab1b0_1
  - freetype=2.12.1=h267a509_2
  - fsspec=2023.12.2=pyhca7485f_0
  - gcc=12.3.0=h8d2909c_2
  - gcc_impl_linux-64=12.3.0=he2b93b0_3
  - gcc_linux-64=12.3.0=h76fc315_2
  - gflags=2.2.2=he1b5a44_1004
  - glog=0.6.0=h6f12383_0
  - gmock=1.14.0=ha770c72_1
  - gtest=1.14.0=h00ab1b0_1
  - gxx=12.3.0=h8d2909c_2
  - gxx_impl_linux-64=12.3.0=he2b93b0_3
  - gxx_linux-64=12.3.0=h8a814eb_2
  - icu=73.2=h59595ed_0
  - identify=2.5.33=pyhd8ed1ab_0
  - importlib-metadata=7.0.0=pyha770c72_0
  - importlib_metadata=7.0.0=hd8ed1ab_0
  - iniconfig=2.0.0=pyhd8ed1ab_0
  - jinja2=3.1.2=pyhd8ed1ab_1
  - kernel-headers_linux-64=2.6.32=he073ed8_16
  - keyutils=1.6.1=h166bdaf_0
  - krb5=1.21.2=h659d440_0
  - lcms2=2.16=hb7c19ff_0
  - ld_impl_linux-64=2.40=h41732ed_0
  - lerc=4.0.0=h27087fc_0
  - libabseil=20230802.1=cxx17_h59595ed_0
  - libarrow=14.0.1=hfb4d3a9_10_cpu
  - libarrow-acero=14.0.1=h59595ed_10_cpu
  - libarrow-dataset=14.0.1=h59595ed_10_cpu
  - libarrow-flight=14.0.1=h120cb0d_10_cpu
  - libarrow-flight-sql=14.0.1=h61ff412_10_cpu
  - libarrow-gandiva=14.0.1=hacb8726_10_cpu
  - libarrow-substrait=14.0.1=h61ff412_10_cpu
  - libblas=3.9.0=20_linux64_openblas
  - libbrotlicommon=1.1.0=hd590300_1
  - libbrotlidec=1.1.0=hd590300_1
  - libbrotlienc=1.1.0=hd590300_1
  - libcblas=3.9.0=20_linux64_openblas
  - libcrc32c=1.1.2=h9c3ff4c_0
  - libcublas=12.0.1.189=hd3aeb46_3
  - libcudf=24.02.00a129=cuda12_231215_g2cb8f3da3a_129
  - libcufft=11.0.0.21=hd3aeb46_2
  - libcufile=1.5.0.59=hd3aeb46_1
  - libcufile-dev=1.5.0.59=hd3aeb46_1
  - libcurand=10.3.1.50=hd3aeb46_1
  - libcurl=8.5.0=hca28451_0
  - libcusolver=11.4.2.57=hd3aeb46_2
  - libcusparse=12.0.0.76=hd3aeb46_2
  - libdeflate=1.19=hd590300_0
  - libedit=3.1.20191231=he28a2e2_2
  - libev=4.33=hd590300_2
  - libevent=2.1.12=hf998b51_1
  - libexpat=2.5.0=hcb278e6_1
  - libffi=3.4.2=h7f98852_5
  - libgcc-devel_linux-64=12.3.0=h8bca6fd_103
  - libgcc-ng=13.2.0=h807b86a_3
  - libgfortran-ng=13.2.0=h69a702a_3
  - libgfortran5=13.2.0=ha4646dd_3
  - libgomp=13.2.0=h807b86a_3
  - libgoogle-cloud=2.12.0=h5206363_4
  - libgrpc=1.59.3=hd6c4280_0
  - libiconv=1.17=hd590300_1
  - libjpeg-turbo=3.0.0=hd590300_1
  - libkvikio=24.02.00a=cuda12_231215_gc6e6456_14
  - liblapack=3.9.0=20_linux64_openblas
  - libllvm14=14.0.6=hcd5def8_4
  - libllvm15=15.0.7=hb3ce162_4
  - libnghttp2=1.58.0=h47da74e_1
  - libnsl=2.0.1=hd590300_0
  - libnuma=2.0.16=h0b41bf4_1
  - libnvjitlink=12.0.76=hd3aeb46_2
  - libopenblas=0.3.25=pthreads_h413a1c8_0
  - libparquet=14.0.1=h352af49_10_cpu
  - libpng=1.6.39=h753d276_0
  - libprotobuf=4.24.4=hf27288f_0
  - libre2-11=2023.06.02=h7a70373_0
  - librmm=24.02.00a25=cuda12_231215_gec7b1644_25
  - libsanitizer=12.3.0=h0f45ef3_3
  - libsqlite=3.44.2=h2797004_0
  - libssh2=1.11.0=h0841786_0
  - libstdcxx-devel_linux-64=12.3.0=h8bca6fd_103
  - libstdcxx-ng=13.2.0=h7e041cc_3
  - libthrift=0.19.0=hb90f79a_1
  - libtiff=4.6.0=ha9c0a0a_2
  - libtool=2.4.7=h27087fc_0
  - libutf8proc=2.8.0=h166bdaf_0
  - libuuid=2.38.1=h0b41bf4_0
  - libuv=1.46.0=hd590300_0
  - libwebp-base=1.3.2=hd590300_0
  - libxcb=1.15=h0b41bf4_0
  - libxml2=2.12.3=h232c23b_0
  - libzlib=1.2.13=hd590300_5
  - llvmlite=0.40.1=py310h1b8f574_0
  - locket=1.0.0=pyhd8ed1ab_0
  - lz4=4.3.2=py310h350c4a5_1
  - lz4-c=1.9.4=hcb278e6_0
  - m4=1.4.18=h516909a_1001
  - markdown-it-py=3.0.0=pyhd8ed1ab_0
  - markupsafe=2.1.3=py310h2372a71_1
  - mdurl=0.1.0=pyhd8ed1ab_0
  - msgpack-python=1.0.7=py310hd41b1e2_0
  - ncurses=6.4=h59595ed_2
  - ninja=1.11.1=h924138e_0
  - nodeenv=1.8.0=pyhd8ed1ab_0
  - numba=0.57.1=py310h0f6aa51_0
  - numpy=1.24.4=py310ha4c1d20_0
  - nvcomp=3.0.5=h10b603f_0
  - nvtx=0.2.8=py310h2372a71_1
  - openjpeg=2.5.0=h488ebb8_3
  - openssl=3.2.0=hd590300_1
  - orc=1.9.2=h4b38347_0
  - packaging=23.2=pyhd8ed1ab_0
  - pandas=1.5.3=py310h9b08913_1
  - partd=1.4.1=pyhd8ed1ab_0
  - perl=5.32.1=4_hd590300_perl5
  - pillow=10.1.0=py310h01dd4db_0
  - pip=23.3.1=pyhd8ed1ab_0
  - pkg-config=0.29.2=h36c2ea0_1008
  - platformdirs=4.1.0=pyhd8ed1ab_0
  - pluggy=1.3.0=pyhd8ed1ab_0
  - pre-commit=3.6.0=pyha770c72_0
  - protobuf=4.24.4=py310h620c231_0
  - psutil=5.9.5=py310h2372a71_1
  - pthread-stubs=0.4=h36c2ea0_1001
  - pyarrow=14.0.1=py310hf9e7431_10_cpu
  - pycparser=2.21=pyhd8ed1ab_0
  - pygments=2.17.2=pyhd8ed1ab_0
  - pynvml=11.4.1=pyhd8ed1ab_0
  - pysocks=1.7.1=pyha2e5f31_6
  - pytest=7.4.3=pyhd8ed1ab_0
  - pytest-asyncio=0.23.2=pyhd8ed1ab_0
  - pytest-rerunfailures=13.0=pyhd8ed1ab_0
  - python=3.10.13=hd12c33a_0_cpython
  - python-dateutil=2.8.2=pyhd8ed1ab_0
  - python_abi=3.10=4_cp310
  - pytz=2023.3.post1=pyhd8ed1ab_0
  - pyyaml=6.0.1=py310h2372a71_1
  - rapids-dask-dependency=24.02.00a7=0
  - rdma-core=49.0=hd3aeb46_1
  - re2=2023.06.02=h2873b5e_0
  - readline=8.2=h8228510_1
  - rhash=1.4.4=hd590300_0
  - rich=13.7.0=pyhd8ed1ab_0
  - rmm=24.02.00a25=cuda12_py310_231215_gec7b1644_25
  - s2n=1.4.0=h06160fa_0
  - scikit-build=0.17.6=pyh4af843d_0
  - setuptools=68.2.2=pyhd8ed1ab_0
  - six=1.16.0=pyh6c4a22f_0
  - snappy=1.1.10=h9fff704_0
  - sortedcontainers=2.4.0=pyhd8ed1ab_0
  - spdlog=1.12.0=hd2e6256_2
  - sysroot_linux-64=2.12=he073ed8_16
  - tblib=3.0.0=pyhd8ed1ab_0
  - tk=8.6.13=noxft_h4845f30_101
  - tomli=2.0.1=pyhd8ed1ab_0
  - toolz=0.12.0=pyhd8ed1ab_0
  - tornado=6.3.3=py310h2372a71_1
  - typing-extensions=4.9.0=hd8ed1ab_0
  - typing_extensions=4.9.0=pyha770c72_0
  - tzdata=2023c=h71feb2d_0
  - ucx=1.15.0=h6d2d1ec_2
  - ukkonen=1.0.1=py310hd41b1e2_4
  - urllib3=2.1.0=pyhd8ed1ab_0
  - virtualenv=20.25.0=pyhd8ed1ab_0
  - wheel=0.42.0=pyhd8ed1ab_0
  - xorg-libxau=1.0.11=hd590300_0
  - xorg-libxdmcp=1.1.3=h7f98852_0
  - xyzservices=2023.10.1=pyhd8ed1ab_0
  - xz=5.2.6=h166bdaf_0
  - yaml=0.2.5=h7f98852_2
  - zict=3.0.0=pyhd8ed1ab_0
  - zipp=3.17.0=pyhd8ed1ab_0
  - zstd=1.5.5=hfc55251_0
prefix: /raid/charlesb/miniforge3/envs/ucxx-cuda120
```

Looks like installing `gcc/gxx=11` unblocks the failures (is this what's getting pulled in CI during the builds?), though @pentschev suggested this could be a cmake issue.

<!-- Edit the body of your new issue then click the ✓ "Create Issue" button in the top right of the editor. The first line will be the issue title. Assignees and Labels follow after a blank line. Leave an empty line before beginning the body of the issue. -->

--- a/LICENSE
+++ b/LICENSE
@@ -1,6 +1,6 @@
 BSD 3-Clause License
 
-Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are met:

--- a/cpp/include/ucxx/constructors.h
+++ b/cpp/include/ucxx/constructors.h
@@ -1,13 +1,13 @@
 /**
- * SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #pragma once
 
 #include <memory>
 #include <string>
-#include <vector>
 
+#include <ucxx/component.h>
 #include <ucxx/request_data.h>
 #include <ucxx/typedefs.h>
 

--- a/cpp/include/ucxx/request_am.h
+++ b/cpp/include/ucxx/request_am.h
@@ -1,5 +1,5 @@
 /**
- * SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #pragma once

--- a/cpp/include/ucxx/request_am.h
+++ b/cpp/include/ucxx/request_am.h
@@ -90,7 +90,7 @@ class RequestAm : public Request {
     RequestCallbackUserFunction callbackFunction,
     RequestCallbackUserData callbackData);
 
-  virtual void populateDelayedSubmission();
+  void populateDelayedSubmission() override;
 
   /**
    * @brief Create and submit an active message send request.

--- a/cpp/python/include/ucxx/python/constructors.h
+++ b/cpp/python/include/ucxx/python/constructors.h
@@ -1,10 +1,12 @@
 /**
- * SPDX-FileCopyrightText: Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-FileCopyrightText: Copyright (c) 2023-2024, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #pragma once
 
 #include <memory>
+
+#include <Python.h>
 
 #include <ucxx/buffer.h>
 

--- a/cpp/python/src/python_future.cpp
+++ b/cpp/python/src/python_future.cpp
@@ -1,5 +1,5 @@
 /**
- * SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #include <memory>

--- a/cpp/python/src/python_future.cpp
+++ b/cpp/python/src/python_future.cpp
@@ -4,6 +4,7 @@
  */
 #include <memory>
 #include <stdexcept>
+#include <utility>
 
 #include <Python.h>
 

--- a/cpp/src/request_am.cpp
+++ b/cpp/src/request_am.cpp
@@ -145,7 +145,7 @@ ucs_status_t RequestAm::recvCallback(void* arg,
       // recvAmMessage.callback(nullptr, UCS_ERR_UNSUPPORTED);
       // return UCS_ERR_UNSUPPORTED;
 
-      ucxx_trace_req("No allocator registered for memory type %lu, falling back to host memory.",
+      ucxx_trace_req("No allocator registered for memory type %u, falling back to host memory.",
                      allocatorType);
       allocatorType = UCS_MEMORY_TYPE_HOST;
     }
@@ -292,7 +292,7 @@ void RequestAm::populateDelayedSubmission()
       ucxx_trace_req_f(_ownerString.c_str(),
                        _request,
                        _operationName.c_str(),
-                       "buffer %p, size %lu, memoryType: %lu, future %p, future handle %p, "
+                       "buffer %p, size %lu, memoryType: %u, future %p, future handle %p, "
                        "populateDelayedSubmission",
                        buffer,
                        length,
@@ -303,7 +303,7 @@ void RequestAm::populateDelayedSubmission()
       ucxx_trace_req_f(_ownerString.c_str(),
                        _request,
                        _operationName.c_str(),
-                       "buffer %p, size %lu, memoryType: %lu, populateDelayedSubmission",
+                       "buffer %p, size %lu, memoryType: %u, populateDelayedSubmission",
                        buffer,
                        length,
                        memoryType);

--- a/cpp/src/request_am.cpp
+++ b/cpp/src/request_am.cpp
@@ -1,5 +1,5 @@
 /**
- * SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #include <cstdio>

--- a/cpp/src/request_tag_multi.cpp
+++ b/cpp/src/request_tag_multi.cpp
@@ -1,5 +1,5 @@
 /**
- * SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #include "ucxx/delayed_submission.h"

--- a/cpp/src/request_tag_multi.cpp
+++ b/cpp/src/request_tag_multi.cpp
@@ -273,8 +273,8 @@ void RequestTagMulti::recvCallback(ucs_status_t status)
       ucxx_trace_req(
         "RequestTagMulti::recvCallback failed receiving header with status %d (%s), multi request: "
         "%p, "
-        "tag: 0x%lx",
-        "tagMask: 0x%lx",
+        "tag: 0x%lx, "
+        "tagMask: 0x%lx,",
         status,
         ucs_status_string(status),
         this,

--- a/cpp/src/request_tag_multi.cpp
+++ b/cpp/src/request_tag_multi.cpp
@@ -2,8 +2,6 @@
  * SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
-#include "ucxx/delayed_submission.h"
-#include <functional>
 #include <memory>
 #include <mutex>
 #include <string>
@@ -274,7 +272,7 @@ void RequestTagMulti::recvCallback(ucs_status_t status)
         "RequestTagMulti::recvCallback failed receiving header with status %d (%s), multi request: "
         "%p, "
         "tag: 0x%lx, "
-        "tagMask: 0x%lx,",
+        "tagMask: 0x%lx",
         status,
         ucs_status_string(status),
         this,

--- a/cpp/src/utils/sockaddr.cpp
+++ b/cpp/src/utils/sockaddr.cpp
@@ -1,5 +1,5 @@
 /**
- * SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #include <arpa/inet.h>

--- a/cpp/src/utils/sockaddr.cpp
+++ b/cpp/src/utils/sockaddr.cpp
@@ -56,8 +56,8 @@ void sockaddr_get_ip_port_str(const struct sockaddr_storage* sockaddr,
       inet_ntop(AF_INET6, &addr_in6->sin6_addr, ip_str, max_str_size);
       snprintf(port_str, max_str_size, "%u", ntohs(addr_in6->sin6_port));
     default:
-      ip_str   = const_cast<char*>(reinterpret_cast<const char*>("Invalid address family"));
-      port_str = const_cast<char*>(reinterpret_cast<const char*>("Invalid address family"));
+      snprintf(ip_str, max_str_size, "Invalid address family");
+      snprintf(port_str, max_str_size, "Invalid address family");
   }
 }
 

--- a/cpp/src/utils/sockaddr.cpp
+++ b/cpp/src/utils/sockaddr.cpp
@@ -51,13 +51,16 @@ void sockaddr_get_ip_port_str(const struct sockaddr_storage* sockaddr,
       addr_in = reinterpret_cast<decltype(addr_in)>(sockaddr);
       inet_ntop(AF_INET, &addr_in->sin_addr, ip_str, max_str_size);
       snprintf(port_str, max_str_size, "%u", ntohs(addr_in->sin_port));
+      break;
     case AF_INET6:
       addr_in6 = reinterpret_cast<decltype(addr_in6)>(sockaddr);
       inet_ntop(AF_INET6, &addr_in6->sin6_addr, ip_str, max_str_size);
       snprintf(port_str, max_str_size, "%u", ntohs(addr_in6->sin6_port));
+      break;
     default:
       snprintf(ip_str, max_str_size, "Invalid address family");
       snprintf(port_str, max_str_size, "Invalid address family");
+      break;
   }
 }
 

--- a/cpp/src/worker.cpp
+++ b/cpp/src/worker.cpp
@@ -204,7 +204,11 @@ void Worker::initBlockingProgressMode()
                              }};
 
   err = epoll_ctl(_epollFileDescriptor, EPOLL_CTL_ADD, _workerFileDescriptor, &workerEvent);
-  if (err != 0) throw std::ios_base::failure(std::string("epoll_ctl() returned " + err));
+  if (err != 0) {
+    std::stringstream ss;
+    ss << "epoll_ctl() returned " << err;
+    throw std::ios_base::failure(ss.str());
+  }
 }
 
 bool Worker::arm()

--- a/cpp/src/worker.cpp
+++ b/cpp/src/worker.cpp
@@ -1,5 +1,5 @@
 /**
- * SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #include <functional>

--- a/cpp/src/worker.cpp
+++ b/cpp/src/worker.cpp
@@ -205,9 +205,7 @@ void Worker::initBlockingProgressMode()
 
   err = epoll_ctl(_epollFileDescriptor, EPOLL_CTL_ADD, _workerFileDescriptor, &workerEvent);
   if (err != 0) {
-    std::stringstream ss;
-    ss << "epoll_ctl() returned " << err;
-    throw std::ios_base::failure(ss.str());
+    throw std::ios_base::failure(std::string("epoll_ctl() returned ") + std::to_string(err));
   }
 }
 

